src/test/java/com/softwaremill/react/kafka/JavaConsumerPropertiesTest.java

```diff
@@ -0,0 +1,69 @@
+package com.softwaremill.react.kafka;
+
+
+import kafka.consumer.ConsumerConfig;
+import kafka.serializer.StringDecoder;
+import org.testng.annotations.Test;
+
+import java.util.UUID;
+
+import static junit.framework.Assert.assertEquals;
+
+public class JavaConsumerPropertiesTest {
+
+    private final String uuid = UUID.randomUUID().toString();
+    private final String brokerList = "localhost:9092";
+    private final String zooKeepHost = "localhost:2181";
+    private final String topic = uuid;
+    private final String groupId = uuid;
+
+    @Test
+    public void HandleBaseCase() {
+
+        final ConsumerProperties consumerProperties = new PropertiesBuilder.Consumer()
+                .withZooKeeperHost(zooKeepHost)
+                .withBrokerList(brokerList)
+                .withGroupId(groupId)
+                .withTopic(topic)
+                .withStringDecoder()
+                .build();
+
+        final ConsumerConfig consumerConfig = consumerProperties.toConsumerConfig();
+
+        assertEquals(consumerProperties.topic(), topic);
+        assertEquals(consumerProperties.groupId(), groupId);
+        assertEquals(consumerProperties.decoder().getClass().getSimpleName(), StringDecoder.class.getSimpleName());
+        assertEquals(consumerConfig.clientId(), groupId);
+        assertEquals(consumerConfig.autoOffsetReset(), "smallest");
+        assertEquals(consumerConfig.offsetsStorage(), "zookeeper");
+        assertEquals(consumerConfig.consumerTimeoutMs(), 1500);
+        assertEquals(consumerConfig.dualCommitEnabled(), false);
+    }
+
+    @Test
+    public void HandleKafkaStorage() {
+
+        final ConsumerProperties consumerProperties = new PropertiesBuilder.Consumer()
+                .withZooKeeperHost(zooKeepHost)
+                .withBrokerList(brokerList)
+                .withGroupId(groupId)
+                .withTopic(topic)
+                .withStringDecoder()
+                .build()
+                .readFromEndOfStream()
```

Comment:
It might be nicer to move these into the builder - so the builder actually sets these before returning the `ConsumerProperties`.
I'm assuming these can't be set when constructing the properties which is why they're being set afterwards?